### ISSUE TITLE
Releasing OpenMVS under GPL?

### ISSUE THREAD

**Issue by ahmadyan** [cdcseacave/openMVS#524]
Followup on issues/268

Is it possible to release openMVS under a less restrictive license than AGPL, e.g. GPL?
We would like to use and contribute to the openMVS pipeline. But the AGPL license is very restrictive for us. 

Thanks

**Comment by cdcseacave** [comment]
I'm not an expert in licenses. Can you pls explain the difference between them?

**Comment by pmoulon** [comment]
Actually AGPL is protecting the project better than GPL.
See https://choosealicense.com/licenses/gpl-3.0/ vs. https://choosealicense.com/licenses/agpl-3.0/
AGPL-> Protect the project towards running a modified version on servers.

Can you list what is problematic for you with the AGPL, an example to understand what is blocking you would be nice?

**Comment by cdcseacave** [comment]
Thank you for the explanation. In this case I have trouble understanding your intentions. On one hand you say you can not run modified version on your internal servers, but on the other you want to contribute to OpenMVS with your changes. So my understanding is as soon as you make your changes part of OpenMVS, you are free to run it on your internal servers. Pls explain what is blocking you?

**Comment by ahmadyan** [comment]
I agree that AGPL does supports commercial products and does protect the open-source projects better than  GPL. However, the primary risk presented by AGPL is that any product or service that depends on AGPL-licensed code, or includes anything copied or derived from AGPL-licensed code, may be subject to the virality of the AGPL license. This viral effect requires that the complete corresponding source code of the product or service be released to the world under the AGPL license. This is triggered if the product or service can be accessed over a remote network interface, so it does not even require that the product or service is actually distributed. 

In other words, If we use AGPL-license code such as OpenMVS on our corp cloud, and some of those services (e.g. Gmail) accidentally link against openMVS the company has to also open-source them since they are running on the same cloud. That's why our  lawyers do not like AGPL license. I also like to add that we are fine with open-sourcing the projects that link/use the OpenMVS library (see https://github.com/google/mediapipe/ where we open-source our projects in my team).

**Comment by cdcseacave** [comment]
First of all I want to make clear that my intent is to find a solution here to make everybody happy: protect everyone interest.

With that in mind, my understanding of GPL is that it allows anybody to modify and use the GPL licensed source code, even commercially, as long as the user makes public any changes made to that source code. However, the release of the changes are demanded ONLY if the software is *distributed*. *Distributed* software does not include usage of that software by clients over the internet/network. In other words any cloud service that uses GPL parts does not have to opensource the changes he made to that piece of code, even if used commercially.

AGPL differs from GPL exactly at this point, otherwise being identical, including the *viral effect*: in AGPL the concept of *distribute* includes cloud computing services. Since these days almost everything is cloud based, and since my intention here is to encourage a community developing *open-sourced* state-of-the-art photogrammetry software, this point in the license is important for me and the reason for which I chose AGPL.

Your intentions seem to completely align with AGPL as well: use OpenMVS freely, but publishing all the changes made to it. Same as GPL, my understanding is AGPL does not require that all components to be open sourced, but only the AGPL module, with the only addition that this includes cloud computing. So your last paragraph might not be accurate: any service linked somehow to OpenMVS does NOT need to be made open-sourced, but ONLY OpenMVS itself needs to remain open-sourced. Can you pls double check this and confirm?

**Comment by ahmadyan** [comment]
Thanks for your thorough response. 

I'm working at a research org, and my intention is to distribute/open-source the final research product (usually through github.com/google/mediapipe/) regardless of us depending on OpenMVS or not. That being said, I can't ask my company to open-source their other projects because they both will be running on the same infra (in AGPL terms, they both will be distributed and subject to AGPL's virality clause). So _there is a blanket ban on AGPL over here_. Unofficially, the whole reason AGPL exists is that internet companies don't use AGPL based software for free (example: MongoDB) without paying the license. As long as OpenMVS uses AGPL, I cannot use that. 

I'm asking to either change the license of OpenMVS, or give us options to license it.

**Comment by cdcseacave** [comment]
As I said above, AGPL does not impose all software connected to OpenMVS to be also AGPL, so your fear is not founded. If you are not convinced, pls share the details of your research org, and I'll be happy to license OpenMVS to you only with a special license that allows only research use-cases.

**Comment by ahmadyan** [comment]
A license would be great! Thank you.
We are the Perception team at Google research. https://research.google/teams/perception/
The library that we are working on is https://github.com/google/mediapipe/

**Comment by cdcseacave** [comment]
Nice to meet you! Pls send me your email so that I can forward you the license agreement. It will be the standard GPLv3 license with the additional note that it will be used for research only.

**Comment by cdcseacave** [comment]
Closing as stale. The codebase has been substantially updated on the \develop\ branch. Please reopen with reproduction steps if the issue persists on the latest code.
